groups/service.go

```diff
@@ -183,7 +183,7 @@ func (svc service) RetrieveGroupHierarchy(ctx context.Context, session smqauthn.
 		return HierarchyPage{}, errors.Wrap(svcerr.ErrViewEntity, err)
 	}
 	hids := svc.getGroupIDs(hp.Groups)
-	ids, err := svc.filterAllowedGroupIDsOfUserID(ctx, session.DomainUserID, "read_permission", hids)
+	ids, err := svc.filterAllowedGroupIDsOfUserID(ctx, session.DomainID, session.UserID, hids)
 	if err != nil {
 		return HierarchyPage{}, errors.Wrap(svcerr.ErrViewEntity, err)
 	}
```

Comment:
We should use different approach, This is legacy way which is inefficient and will not scale.  
We should follow the approach used for listing groups.
I will come up with SQL, or idea how to approach this..